tests/Provider/ConcreteProviderTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Yiisoft\EventDispatcher\Tests\Provider;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\EventDispatcher\Provider\ConcreteProvider;
+use Yiisoft\EventDispatcher\Tests\Event\ClassInterface;
+use Yiisoft\EventDispatcher\Tests\Event\ClassItself;
+use Yiisoft\EventDispatcher\Tests\Event\Event;
+use Yiisoft\EventDispatcher\Tests\Event\ParentClass;
+use Yiisoft\EventDispatcher\Tests\Event\ParentInterface;
+
+use function array_slice;
+
+class ConcreteProviderTest extends TestCase
+{
+    /**
+     * @var ConcreteProvider
+     */
+    private ConcreteProvider $concreteProvider;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->concreteProvider = new ConcreteProvider();
```

Comment:
Please don't use private property if there is no need

Reply:
You mean I should instantiate the `ConcreteProvider` in every test, right?

Reply:
Yes, I do

Reply:
I don't see the benefit; instantiating at one place has several benefits: it removes the responsibility from the test case, and makes potential future refactors easier. I'd like input from other reviewers on this.

Reply:
Tests in general should be able to be executed in random order, i.e. no shared state should present. In case of sharing state one test may affect another producing side-effects.

If initialization of the object is tricky, it could be moved to a method but in this case it's simple.

Reply:
I'll fix it.

Reply:
Ok seems legit; thanks for the fix!